src/hypothesis/strategies.py

```diff
@@ -711,6 +711,29 @@ def text(
 
 @cacheable
 @defines_strategy
+def text_matching(regex):
+    """Return strategy that generates strings that match given regex.
+
+    Regex can be either a string or compiled regex (through `re.compile()`).
+
+    You can use regex flags (such as `re.IGNORECASE`, `re.DOTALL` or `re.UNICODE`)
```

Comment:
formatting: double-backticks for code literals in RST; single-backticks is a cross-reference.

More generally, you can use 

    :const:`python:re.IGNORECASE`

to link to the stdlib docs on that thing and I would suggest doing so.